docs/deployment.rst

```diff
@@ -0,0 +1,144 @@
+
+* The backup server: `backup01.srv.local`
+* The clients:
+  * John Doe's desktop: `johndoe.clnt.local`
+  * Webserver 01: `web01.srv.local`
+  * Application server 01: `app01.srv.local`
+
+User and group
+--------------
+
+The storage server needs to have only one UNIX user for all the clients.
+Recommended user and group and additional settings:
+
+* User: `backup`
+* Group: `backup`
+* Shell: e.g `/bin/bash` which is capable to run the `borg serve` command
+
+The clients probably will initiate a backup from the root user to catch all
+users, groups and permissions (e.g when backupping `/home`).
+
+Folders
+-------
+
+The following folder tree layout is suggested on the storage server:
+
+* User home directory, /home/backup
+* Repositories path (storage pool): /home/backup/repos
+* Clients restricted paths: `/home/backup/repos/<client fqdn>`
+  * johndoe.clnt.local: `/home/backup/repos/johndoe.clnt.local`
+  * web01.srv.local: `/home/backup/repos/web01.srv.local`
+  * app01.srv.local: `/home/backup/repos/app01.srv.local`
+
+Restrictions
+------------
+
+We need to restric the clients into their paths. Borg needs to be instructed to
+enforce this: ``borg serve --restrict-path /home/backup/repos/<client fqdn>``
+
+Clients can have one or more allowed ssh keys. Albeit recommended is to only use
+one per client. We add keys for ``johndoe.clnt.local``, ``web01.srv.local`` and
+``app01.srv.local``. Only they run under the user ``backup@backup01.srv.local``. A option
+per on-line of ``authorized_keys`` is added to distinguish between the clients:
+
+::
+
+  command="cd /home/backup/repos/<client fqdn>;
+           borg serve --restrict-path /home/backup/repos/<client fqdn>",
+           no-port-forwarding,no-X11-forwarding,no-pty <keytype> <key> <host>
+
+**NOTE** The command above need to be compacted into a single line when present in
+``$HOME/.ssh/authorized_keys``
+
+* Change working directory: ``cd /home/backup/repos/<client fqdn>``
+* Run ``borg serve`` with restriction: ``borg serve --restrict-path /home/backup/repos/<client fqdn>``
+* Restrict ssh channel: ``no-port-forwarding,no-X11-forwarding,no-pty``
+
+Client
+------
+
+The client needs to initialize the `pictures` repository like this:
+
+`borg init backup@backup01.srv.local:pictures`
+
+Or with the full path (should actually never be used, as only for demonstrational purposes).
+The server should automatic change the current working directory to the `<client fqdn>` folder.
+
+`borg init backup@backup01.srv.local:/home/backup/repos/johndoe.clnt.local/pictures`
+
+When `johndoe.clnt.local` tries to access a not restricted path the following error is raised.
+John Doe tries to backup into the Web 01 path: `borg init backup@backup01.srv.local:/home/backup/repos/web01.srv.local/pictures`
+
+::
+
+  ~~~ SNIP ~~~
+  Remote: borg.remote.PathNotAllowed: /home/backup/repos/web01.srv.local/pictures
+  ~~~ SNIP ~~~
+  Repository path not allowed
+
+Ansible
+-------
```

Comment:
Yes because it automates the process over SSH from a management machine to satisfy the configuration.